interpreter/src/objects.rs

```diff
@@ -482,6 +482,38 @@ impl<'a> Value {
                         .cloned()
                         .unwrap_or(Value::Null)
                         .into(),
+                    (Value::Map(map), Value::Bool(property)) => map
+                        .map
+                        .get(&property.into())
+                        .cloned()
+                        .unwrap_or(Value::Null)
+                        .into(),
+                    (Value::Map(map), Value::Int(property)) => map
+                        .map
+                        .get(&property.into())
+                        .cloned()
+                        .or_else(|| {
+                            // Check for matching unsinged int indexes too.
+                            let Ok(index) = u64::try_from(property) else {
+                                return None;
+                            };
+                            map.map.get(&index.into()).cloned()
+                        })
+                        .unwrap_or(Value::Null)
+                        .into(),
+                    (Value::Map(map), Value::UInt(property)) => map
+                        .map
+                        .get(&property.into())
+                        .cloned()
+                        .or_else(|| {
+                            // Check for matching singed int indexes too.
+                            let Ok(index) = i64::try_from(property) else {
+                                return None;
+                            };
+                            map.map.get(&index.into()).cloned()
+                        })
+                        .unwrap_or(Value::Null)
+                        .into(),
```

Comment:
I think I'm good with this. Was just doing some experimenting to see if there was a good way to reduce the duplication, but I'm not sure whether something like this is appreciably different since it just moves the double-checking somewhere else. Let me noodle on this a little longer and see if anything else comes to mind.

```rust
(Value::Map(map), Value::Int(v)) => {
    map.get_numeric(v.into()).unwrap_or(Value::Null).into()
}
(Value::Map(map), Value::UInt(v)) => {
    map.get_numeric(v.into()).unwrap_or(Value::Null).into()
}
```

Reply:
How do you feel about the update I just pushed? I factored out as much as I could (plus it feels appropriate for a public Map type to have a `get` function), so the duplication is about as minimal as Rust enums allow now without resorting to pulling in the `num` crate and adding a bunch of boilerplate to allow working with keys as generics or something